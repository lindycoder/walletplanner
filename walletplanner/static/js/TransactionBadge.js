import React from 'react';
import PropTypes from 'prop-types';
import CurrencyInput from 'react-currency-input';


class TransactionBadge extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <span className="badge badge-light transaction-badge-amount">{this.formatAmount(this.props.transaction.amount)}</span>
        );
    }

    formatAmount(amount) {
        return `${(amount / 100).toFixed(2)}$`
    }
};


TransactionBadge.propTypes = {
  transaction: PropTypes.any.isRequired
};

export default TransactionBadge;