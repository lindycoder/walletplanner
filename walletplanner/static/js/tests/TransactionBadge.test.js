import React from 'react';
import { shallow } from 'enzyme';
import TransactionBadge from '../TransactionBadge';

describe('TransactionBadge', () => {
    let props;
    let wrapper;

    beforeEach(() => {
        props = {
            transaction: {
               "amount": 1000,
               "description": "desc1",
               "category": "cat1"
            }
        };
        wrapper = shallow(<TransactionBadge {...props} />);
    });

    it('should render', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the amount correctly', () => {
        expect(wrapper.find('.transaction-badge-amount').first().text()).toBe("10.00$")
    });
});
