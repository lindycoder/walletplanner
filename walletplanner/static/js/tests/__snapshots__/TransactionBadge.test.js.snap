// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionBadge should render 1`] = `
<span
  className="badge badge-light transaction-badge-amount"
>
  10.00$
</span>
`;
